### PR TITLE
feat(post): added tests for post in dao and graphql

### PR DESCRIPTION
**Pull request by saurabh-wednesday**
### Ticket Link
---------------------------------------------------

### Related Links
---------------------------------------------------

### Description
---------------------------------------------------
added test for posts

### Steps to Reproduce / Test
---------------------------------------------------

### Request
---------------------------------------------------

### Response
---------------------------------------------------

### CODE REVIEW

File: daos/authors.go [hunk 1]
```diff
@@ -58,6 +58,6 @@ func FetchAllAuthors(ctx context.Context) (models.AuthorSlice, error) {
 
 // FetchAuthorByID returns the Author associated with the id
 func FetchAuthorByID(ID int, ctx context.Context) (*models.Author, error) {
-	contextExecutor := getContextExecutor(nil)
+	contextExecutor := getContextExecutor(&sql.Tx{})
```
Comment (rajatdangat-ws) [inline daos/authors.go]: Why are we not passing nil here?

File: daos/post.go [hunk 2]
```diff
@@ -11,26 +11,25 @@ import (
...
 }
 
-// updatePostTx creates a transaction for the updation of post
-func updatePostTx(post *models.Post, ctx context.Context, tx *sql.Tx) (models.Post, error) {
-	contextExecutor := getContextExecutor(tx)
+// UpdatePostTx creates a transaction for the updation of post
+func UpdatePostTx(post *models.Post, ctx context.Context, tx *sql.Tx) (models.Post, error) {
 
-	_, err := post.Update(ctx, contextExecutor, boil.Infer())
+	_, err := post.Update(ctx, boil.GetContextDB(), boil.Infer())
```
Comment (rajatdangat-ws) [inline daos/post.go]: why have we removed `getContextExecutor`?

File: resolver/post_query.resolvers_test.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+		ctx context.Context
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    *gqlmodels.PostPayload
+		wantErr bool
+	}{
+		// TODO: Add test cases.
```
Comment (rajatdangat-ws) [inline resolver/post_query.resolvers_test.go]: TODO remaining?

File: daos/authors.go (hunk 1 above)
Comment (saurabh-wednesday) [reply daos/authors.go]: added them to verify some sql.Tx properties, reverted back to nil

File: daos/post.go (hunk 2 above)
Comment (saurabh-wednesday) [reply daos/post.go]: reverted back to contextExecutor

**Review by rajatdangat-ws — APPROVED** [review]

**Pull request merged**